pages/preso.tsx

```diff
@@ -5,34 +5,40 @@ import { PresoForm } from '@types'
 import { NextPage } from 'next'
 import Head from 'next/head'
 import { useRouter } from 'next/router'
+import {validatePresos} from "validation/presos"
 
 const Preso: NextPage = () => {
   const router = useRouter()
   const { session } = useSupabase()
 
   const onSubmit = async (values: PresoForm) => {
+    const { title, url, eventName, eventLocation } = values;
     try {
-      const response = await fetch('/api/preso', {
-        method: 'POST',
-        body: JSON.stringify({
-          ...values,
-          userId: session!.user?.id
-        } as PresoForm)
-      })
-      const json = await response.json()
-      router.replace(`/qr?preso=${encodeURIComponent(json.presoShortCode)}`)
+      const isPresosvalid = await validatePresos.validate({ title, url, eventName, eventLocation });
```

Comment:
It would be better to pass the validation schema in the form since formik providers are a prop for it. 
```tsx
<Formik
      initialValues={
        { url: '', eventName: '', title: '', eventLocation: '' } as PresoForm
      }
      onSubmit={props.onSubmit}
      validate={onValidate}
      validationSchema={validatePresos}
    >
```